addons/account/models/sequence_mixin.py

```diff
@@ -41,6 +41,19 @@ def init(self):
                     field=self._sequence_field,
                 ))
 
+    def write(self, vals):
+        for record in self:
+            date_f = record._sequence_date_field
+            if date_f in vals and record[date_f] != vals[date_f]:
+                # Ensure consistency between date and sequence
+                format_values = record._get_sequence_format_param(record[record._sequence_field] or '')[1]
+                if (
+                    format_values['year'] and format_values['year'] != fields.Date.to_date(vals[date_f]).year
+                    or format_values['month'] and format_values['month'] != fields.Date.to_date(vals[date_f]).month
+                ):
+                    raise ValidationError(_("The date doesn't match the sequence number."))
```

Comment:
maybe it would be better to better guide the user, what to do with the error. something like
```suggestion
                    raise ValidationError(_("The date doesn't match the sequence number. You might want to remove the sequence before proceeding with the change of the date"))
```